docs/src/writing_good_rules.md

```diff
@@ -1,5 +1,60 @@
 # On writing good `rrule` / `frule` methods
 
+## Use `Type{T}`, not `typeof(T)`, to define rules for constructors
+
+To define an `frule` or `rrule` for a _function_ `foo` we dispatch on the type of `foo`, which is `typeof(foo)`.
+For example, the `rrule` signature would be like:
+
+```julia
+function rrule(::typeof(foo), args...; kwargs...)
+    ...
+    return y, foo_pullback
+end
+```
+
+But to define an `rrule` for a constructor for a  _type_ `T` we need to be careful to dispatch only on `Type{T}`.
+
+For example, the `rrule` signature for a constructor would be like:
+
+```julia
+function rrule(::Type{T}, args...; kwargs...)
+    ...
+    return y, T_pullback
+end
+```
+
+In particular, be careful not to use `typeof(T)` here.
+Because `typeof(T)` is `DataType`, using this to define an `rrule`/`frule` will define an `rrule`/`frule` for all constructors.
+
+You can check which to use with `Core.Typeof`:
```

Comment:
I think more useful than the below part on `Typeof` would be to create a struct that is also a functor and show how to define `rrule`s for both its constructor and it as a functor. Writing functor rules also trips up users, and the two are easily confused.